index.js

```diff
@@ -0,0 +1,259 @@
+const navWrapper = document.querySelector(".nav-wrapper");
+const navToggle = document.querySelector(".mobile-nav-toggle");
+const primaryNavigation = document.querySelector(".primary-navigation");
+
+document.body.addEventListener("click", (e) => {
+  if (
+    !navWrapper.contains(e.target) &&
+    navToggle.attributes[2].nodeValue === "true"
+  ) {
+    closeNav();
+  }
+});
+navWrapper.addEventListener("click", (e) => {
+  if (navToggle.attributes[2].nodeValue === "false") {
+    showNav();
+  } else {
+    closeNav();
+  }
+});
+
+const showNav = () => {
+  navToggle.setAttribute("aria-expanded", true);
+  primaryNavigation.toggleAttribute("opening");
+  primaryNavigation.addEventListener(
+    "animationend",
+    () => {
+      primaryNavigation.toggleAttribute("opened");
+      primaryNavigation.toggleAttribute("opening");
+    },
+    { once: true }
+  );
+};
+
+const closeNav = () => {
+  primaryNavigation.toggleAttribute("closing");
+  navToggle.setAttribute("aria-expanded", false);
+  primaryNavigation.addEventListener(
+    "animationend",
+    () => {
+      primaryNavigation.toggleAttribute("opened");
+      primaryNavigation.toggleAttribute("closing");
+    },
+    { once: true }
+  );
+};
+
+const contactMeButton = document.querySelector("#contact-me");
+const modal = document.querySelector("#modal");
+const closeModalButton = document.querySelector(".modal__close");
+const mobileContactButton = document.querySelector(".button--mobile");
+
+const closeModalAnimation = () => {
+  modal.setAttribute("closing", "");
+
+  modal.addEventListener(
+    "animationend",
+    () => {
+      modal.removeAttribute("closing");
+      modal.close();
+    },
+    { once: true }
+  );
+  document.body.style.overflow = "auto";
+};
+
+contactMeButton.addEventListener("click", () => {
+  modal.showModal();
+  document.body.style.overflow = "hidden";
+});
+
+mobileContactButton.addEventListener("click", () => {
+  modal.showModal();
+});
+
+modal.addEventListener("click", (e) => {
+  e.stopPropagation();
```

Comment:
What was happening if you didn't `stopPropagation`?

Reply:
This is a piece of old code I need to remove. Thank you for reminding me. Before I had the modal, I had the modal as a child of something else, and I was trying to figure out how to get pseudo elements through javascript. But luckily, the backdrop pseudo-element on a dialog tag shows up as part of the dialog itself.